mantidproject/mantid
# mantid-developer metapackage

### Description of work
Adds a new Conda recipe that generates a `mantid-developer` Conda [metapackage](https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/packages.html#metapackages). The metapackage is a replacement for the `mantid-developer-{linux,osx,win}.yml` environment files that were used to create development environments. The major advantage of this is that dependency version numbers only appear in one place. There were still a few instances where the version pinning was different in the developer and packaging environments. Those should all be resolved now.

#### Summary of work
- Consolidate all developer environments into a single metapackage recipe, and share the version pinnings with the other conda recipes.
- Configure the build scripts so that the metapackage is built locally from source and installed from the local channel before every build. Previously the mantid-developer environment was created from the environment files before each build.

Fixes #37627 


#### Further detail of work
- PR builds will build the metapackage before each job run and install the locally built metapackage. This ensures that any changes to the recipe are tested before they are merged.
- A new metapackage will be uploaded along with the other Conda packages to our Conda channel.

### To test:
- All stages in this pipeline build should succeed, and it should publish the mantid-developer metapackage to the `metapackage_test` label:
https://builds.mantidproject.org/job/build_packages_from_branch/880/
- Try to install the metapackage by:
```
mamba create -n mantid-developer-test
mamba activate mantid-developer-test
mamba install -c mantid/label/metapackage_test mantid-developer
```
- Check that the developer documentation changes are good





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
